### PR TITLE
Issue 258: Document and revise parallel_execution_omp

### PR DESCRIPTION
**Pull request by jdgarciauc3m**

### CODE REVIEW

File: include/native/farm.h [hunk 1]
```diff
@@ -57,21 +57,19 @@ void farm(parallel_execution_native & ex, Generator generate_op,
...
   using namespace std;
   using result_type = typename result_of<Generator()>::type;
-  mpmc_queue<result_type> queue{ex.queue_size,ex.lockfree};
+  auto queue = ex.make_queue<result_type>();
 
   vector<thread> tasks;
   for (int i=0; i<ex.concurrency_degree(); ++i) {
     tasks.emplace_back([&](){
-      ex.register_thread();
+      auto manger = ex.thread_manager();
```
Comment (mdolz) [inline include/native/farm.h]: Replace `manger` by `manager`.

File: include/native/parallel_execution_native.h [hunk 2]
```diff
@@ -32,105 +32,236 @@
...
+  while (lock_.test_and_set(memory_order_acquire)) {}
+  auto this_id = this_thread::get_id();
+  auto current = find(begin(ids_), end(ids_), this_id);
+  auto index = distance(begin(ids_), current);
+  lock_.clear(memory_order_release);
+  return index;
+};
+
+/**
+\brief RAII class to mange registration/deregistration pairs.
```
Comment (mdolz) [inline include/native/parallel_execution_native.h]: Replace `mange` by `manage`.

File: include/native/parallel_execution_native.h [hunk 3]
```diff
@@ -32,105 +32,236 @@
...
+  */
+  void set_queue_attributes(int size, queue_mode mode) noexcept {
+    queue_size_ = size;
+    queue_mode_ = mode;
+  }
+
+  /**
+  \brief Makes a communication queue for elements of type T.
+  Constructs a queue using the attributes that can be set via 
+  et_queue_attributes(). The value is returned via move semantics.
```
Comment (mdolz) [inline include/native/parallel_execution_native.h]: Replace `et` by `set`.

File: include/tbb/parallel_execution_tbb.h [hunk 4]
```diff
@@ -30,73 +30,124 @@
...
+/**
+\brief Metafunction that determines if type E is supported in the current build.
+\tparam Execution policy type.
+*/
 template <typename E>
 constexpr bool is_supported();
 
+/**
+\brief Specialization stating that parallel_execution_native is supported.
+This metafunction evaluates to fals if GRPPI_TBB is enabled.
```
Comment (mdolz) [inline include/tbb/parallel_execution_tbb.h]: Replace `fals` by `false`.

**Review by mdolz — DISMISSED** [review]
See comments. Most of them are duplicated from the parallel_execution_tbb pull request, so they may be automatically fixed if you process the other request first.

**Review by Drioastorga — APPROVED** [review]
Ok.

**Review by javidoc00 — APPROVED** [review]

**Comment by jdgarciauc3m** [comment]
Closed issue #258

**Pull request merged**